code/modules/power/cell.dm

```diff
@@ -76,8 +76,9 @@
 // recharge the cell
 /obj/item/stock_parts/cell/proc/give(amount)
 	if(rigged && amount > 0)
-		explode()
-		return 0
+		if(!((locate(/mob/living/simple_animal/pulse_demon) in src) && istype(loc, /obj/machinery/cell_charger)))
+			explode()
+			return 0
```

Comment:
this is a very snowflakey check, any better ways to do this?

Reply:
Perhaps only set rigged when the demon jumps out of a cell? Putting that in forceMove might cover all the cases but I'm not sure if something else might need to check too.